src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -88,10 +88,15 @@ private EventsourcedState RecoveryStarted(long maxReplays)
                         timeoutCancelable.Cancel();
                         if (res.Snapshot != null)
                         {
-                            var snapshot = res.Snapshot;
-                            LastSequenceNr = snapshot.Metadata.SequenceNr;
-                            // Since we are recovering we can ignore the receive behavior from the stack
-                            base.AroundReceive(recoveryBehavior, new SnapshotOffer(snapshot.Metadata, snapshot.Snapshot));
+                            var offer = new SnapshotOffer(res.Snapshot.Metadata, res.Snapshot.Snapshot);
+                            if (base.AroundReceive(recoveryBehavior, offer))
```

Comment:
In scala you can check whether a partial function can handle a message (i.e, `recoveryBehavior.isDefinedAt(offer)`) before actually calling it, but I don't think we can do anything like that in C#, except to actually invoke it.

Reply:
Yes, we cannot do that in C#, but also we cannot allow to do something like this (executing AroundReceive before setting a snapshot) - as this is a change of behavior.

What we could do is to optimistically set `LastSequenceNr`  and call `base.AroundReceive`. Then when returned value was false, we can retroactivelly set `LastSequenceNr` back to the previous value:

```csharp
var seqnr = LastSequenceNr;
LastSequenceNr = res.Snapshot.Metadata.SequenceNr;
if (!base.AroundReceive(recoveryBehavior, offer) 
{
    LastSequenceNr = seqNr;
    Unhandled(offer);
}
```

Reply:
I'll push a fix later today.

Reply:
@Horusiath just in case you missed my last push.